pandas/core/indexes/multi.py

```diff
@@ -324,11 +324,17 @@ def from_arrays(cls, arrays, sortorder=None, names=None):
                    codes=[[0, 0, 1, 1], [1, 0, 1, 0]],
                    names=['number', 'color'])
         """
+        error_msg = "Input must be a list / sequence of array-likes."
         if not is_list_like(arrays):
-            raise TypeError("Input must be a list / sequence of array-likes.")
+            raise TypeError(error_msg)
         elif is_iterator(arrays):
             arrays = list(arrays)
 
+        # Check if elements of array are list-like
```

Comment:
do we fully test this? (test with tuples as well)

Reply:
@jreback: not fully. i've added a test that is basically a cut and paste from another test. could parameterise now or refactor in a follow-on PR. i prefer the later since some other refactoring of the tests may be possible and may detract from the current change.